Program.cs

```diff
@@ -30,47 +27,52 @@ public static void Main(params string[] args)
                 Console.WriteLine("Ditto will still work as a local database.");
             }
 
-            Console.WriteLine("Welcome to Ditto's Task App");
+            Console.WriteLine("Welcome to Ditto's Task App\n");
+
+            string query = "SELECT * FROM tasks WHERE isDeleted == false";
+
+            ditto.Sync.RegisterSubscription(query);
 
-            subscription = ditto.Store["tasks"].Find("!isDeleted").Subscribe();
-            liveQuery = ditto.Store["tasks"].Find("!isDeleted").ObserveLocal((docs, _event) => {
-                tasks = docs.ConvertAll(document => new Task(document));
+            ditto.Store.RegisterObserver(query, (result) =>
+            {
+                tasks = result.Items.ConvertAll(item => Task.JsonToTask(item.JsonString()));
             });
 
-            ditto.Store["tasks"].Find("isDeleted == true").Evict();
+            await ditto.Store.ExecuteAsync("EVICT FROM tasks WHERE isDeleted == true");
 
             ListCommands();
 
+
             while (!isAskedToExit)
             {
-                
+
                 Console.Write("\nYour command: ");
                 string command = Console.ReadLine();
 
                 switch (command)
                 {
-                    
-                    case string s when command.StartsWith("--upsert"):
-                        string taskBody = s.Replace("--upsert ", "");
-                        ditto.Store["tasks"].Upsert(new Task(taskBody, false).ToDictionary());
+                    case string s when command.StartsWith("--insert"):
+                        string taskBody = s.Replace("--insert ", "");
+                        var task = new Task(taskBody, false).ToDictionary();
+                        await ditto.Store.ExecuteAsync("INSERT INTO tasks DOCUMENTS (:task)", new Dictionary<string, object> { { "task", task } });
                         break;
                     case string s when command.StartsWith("--toggle"):
                         string _idToToggle = s.Replace("--toggle ", "");
-                        ditto.Store["tasks"]
-                            .FindById(new DittoDocumentId(_idToToggle))
-                            .Update((mutableDoc) => {
-                                if (mutableDoc == null) return;
-                                mutableDoc["isCompleted"].Set(!mutableDoc["isCompleted"].BooleanValue);
-                            });
+                        await ditto.Store.ExecuteAsync("SELECT * FROM tasks WHERE _id == :id", new Dictionary<string, object> { { "id", _idToToggle } }).ContinueWith((result) =>
+                        {
+                            var task = result.Result.Items[0];
+                            var isCompleted = Task.JsonToTask(task.JsonString()).isCompleted;
+                            ditto.Store.ExecuteAsync("UPDATE tasks SET isCompleted = :toggledValue WHERE _id == :id", new Dictionary<string, object> { { "id", _idToToggle }, { "toggledValue", !isCompleted } });
```

Comment:
Hmm what happens if you don't await this in C#? Cc @busec0

Reply:
`ExecuteAsync` is an asynchronous function that does `UPDATE` in the background. Once the `UPDATE` is done, `RegisterObserver`'s callback will catch the updated result. This way, this action's performance should become better since the process doesn't have to wait on this line.
We should recommend doing `await` only when the action needs to be really awaited to prevent a slow process.

Reply:
> Hmm what happens if you don't await this in C#? 

This is actually a pretty large topic, but a TLDR would be: weird stuff can happen and it should usually be avoided. Using `async-await` won't block the thread. 

If code should be executed in parallel, a pattern to use would be: 

```
var task1 = DoSomethingAync(); 
DoSomethingElseWhileTask1IsStillInProgress(); 
await task1; 
```